diff --git a/ucm2/codecs/qcom-lpass/wsa-macro/SpeakerEnableSeq.conf b/ucm2/codecs/qcom-lpass/wsa-macro/SpeakerEnableSeq.conf
index 488826b..8b2116e 100644
--- a/ucm2/codecs/qcom-lpass/wsa-macro/SpeakerEnableSeq.conf
+++ b/ucm2/codecs/qcom-lpass/wsa-macro/SpeakerEnableSeq.conf
@@ -5,4 +5,6 @@ EnableSequence [
 	cset "name='WSA_RX1 INP0' RX1"
 	cset "name='WSA_COMP1 Switch' 1"
 	cset "name='WSA_COMP2 Switch' 1"
+	cset "name='WSA_RX0 Digital Volume' 84"
+	cset "name='WSA_RX1 Digital Volume' 84"
 ]
